cmd/adapter/main.go

```diff
@@ -272,6 +261,20 @@ func main() {
 	}
 	cmd.WithExternalMetrics(kedaProvider)
 
+	clientset, err := cmd.DiscoveryClient()
+	if err != nil {
+		logger.Error(err, "not able to get Kubernetes version")
+		return
+	}
+	version, err := clientset.ServerVersion()
+	if err != nil {
+		logger.Error(err, "not able to get Kubernetes version")
+		return
+	}
+	kubeVersion := kedautil.NewK8sVersion(version)
```

Comment:
Could you move this to some function/package? for example pkg/k8s

This is the very same code that's used to get ScaleClient : https://github.com/kedacore/keda/blob/c1611482226aae17b8af1a575f1a629f0c912bd1/pkg/k8s/scaleclient.go#L36-L49

Reply:
sure, I'll do

Reply:
I have checked it better and I'm not sure if this is worth because we use `clientset` (in scaleclient.go) a few lines above, I'm not sure if we will improve the code or we are doing overengineering 🤔 
Maybe a function `GetK8sVersionFromMgr` and another `GetK8sVersionFromAdapater` but I'm still not sure, wdyt?

Reply:
okay, let's keep it this way